File: editorial/chap5-7/prob5-7-4.cpp
#include <iostream>
#include <algorithm>
using namespace std;

long long N;
long long X[200009], Y[200009];

int main() {
	// 入力
	cin >> N;
	for (int i = 1; i <= N; i++) cin >> X[i] >> Y[i];
	
	// 配列をソートする
	sort(X + 1, X + N + 1);
	sort(Y + 1, Y + N + 1);
	
	// パーツ 1 の答え（x 座標の差の絶対値の総和）
	long long Part1 = 0;
	for (int i = 1; i <= N; i++) Part1 += X[i] * (-N + 2LL * i - 1LL);
	
	// パーツ 2 の答え（y 座標の差の絶対値の総和）
	long long Part2 = 0;
	for (int i = 1; i <= N; i++) Part2 += Y[i] * (-N + 2LL * i - 1LL);
	
	// 出力
	cout << Part1 + Part2 << endl;
	return 0;
}

File: editorial/chap5-7/prob5-7-4.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 入力
		Scanner sc = new Scanner(System.in);
		int N = sc.nextInt();
		long[] X = new long[N];
		long[] Y = new long[N];
		for (int i = 1; i <= N; i++) {
			X[i - 1] = sc.nextLong();
			Y[i - 1] = sc.nextLong();
		}
		
		// 配列 X, Y をソートする
		Arrays.sort(X);
		Arrays.sort(Y);
		
		// パーツ 1 の答え（x 座標の差の絶対値の総和）
		long Part1 = 0;
		for (int i = 1; i <= N; i++) Part1 += X[i - 1] * (-N + 2L * i - 1L);
		
		// パーツ 2 の答え（y 座標の差の絶対値の総和）
		long Part2 = 0;
		for (int i = 1; i <= N; i++) Part2 += Y[i - 1] * (-N + 2L * i - 1L);
		
		// 出力
		System.out.println(Part1 + Part2);
	}
}